GenderPayGap.WebUI/Search/AutoCompleteSearchService.cs

```diff
@@ -0,0 +1,169 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using GenderPayGap.Core;
+using GenderPayGap.WebUI.Models.Search;
+using GenderPayGap.WebUI.Search.CachedObjects;
+
+namespace GenderPayGap.WebUI.Search
+{
+
+    internal class AutoCompleteOrganisation
+    {
+        public string EncryptedId { get; set; }
+        public List<AutoCompleteOrganisationName> Names { get; set; }
+        public List<double> Ranks { get; set; }
+    }
+
+    internal class AutoCompleteOrganisationName
+    {
+        public string Name { get; set; }
+        public List<double> Ranks { get; set; }
+    }
+
+
+    public class AutoCompleteSearchService
+    {
+
+        public List<SuggestEmployerResult> Search(string query)
+        {
+            query = query.Trim().ToLower();
+
+            List<string> searchTerms = ExtractSearchTermsFromQuery(query);
+
+            List<SearchCachedOrganisation> allOrganisations = SearchRepository.CachedOrganisations;
+
+            List<SearchCachedOrganisation> matchingOrganisations = GetMatchingOrganisations(allOrganisations, searchTerms);
+
+            List<AutoCompleteOrganisation> orderedOrganisations = CalculateRankings(matchingOrganisations, searchTerms, query);

```

Comment:
I assume the `CalculateRankings` also orders the results, based on the variable that you're using. Would it be worth renaming the method name to make this more clear? Something like `OrderOrganisationsByRanking`